### PR TITLE
support cast decimal to decimal

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

part of https://github.com/apache/arrow-rs/issues/1043

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/src/compute/kernels/cast.rs [hunk 1]
```diff
@@ -1908,6 +1945,14 @@ mod tests {
     use crate::{buffer::Buffer, util::display::array_value_to_string};
     use num::traits::Pow;
 
+    #[test]
+    fn test_cast_decimal_to_decimal() {
+        let input_type = DataType::Decimal(20, 3);
+        let output_type = DataType::Decimal(20, 4);
+        assert!(can_cast_types(&input_type, &output_type));
+        // let array = vec![Some(1123456), Some(2123456), Some(3123456), None];
```
Comment (alamb) [inline arrow/src/compute/kernels/cast.rs:1953]: looks like we just need some tests here, and we'll be all set 👍

**Review by alamb — COMMENTED** [review]
@liukun4515  I looked at this PR and the code looks good. I think all it needs now is some tests and a rebase and it would be good to go.

I plan to create an arrow 6.5.0 release candidate tomorrow (Thursday) -- let me know if you think we should aim to get this PR merged before doing so.

**Comment by liukun4515** [comment]
> @liukun4515 I looked at this PR and the code looks good. I think all it needs now is some tests and a rebase and it would be good to go.
> 
> I plan to create an arrow 6.5.0 release candidate tomorrow (Thursday) -- let me know if you think we should aim to get this PR merged before doing so.

I rebased this branch with master and added some test cases.
If it looks good for you, you can merge it.
Thank.

**Review by houqp — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @liukun4515 ❤️

**Pull request merged**